struts-pwn.py

```diff
@@ -303,25 +316,32 @@ def main(url=url, usedlist=usedlist, cmd=cmd, do_exploit=do_exploit):
             print('Error: There was an error in reading list file.')
             print("Exception: " + str(e))
             exit(1)
-        for url in URLs_List:
+
+        plock = threading.Lock()
+        def worker(url):
             if not do_exploit:
                 result = check(url)
                 output = '[*] Status: '
                 if result is True:
                     output += 'Vulnerable!'
                 else:
                     output += 'Not Affected.'
-                print(output)
+                with plock:
+                    print(output)
             else:
                 exploit(url, cmd)
-                print("[$] Request sent.")
-                print("[.] If the host is vulnerable, the command will be executed in the background.")
+                with plock:
+                    print("[$] Request sent.")
+                    print("[.] If the host is vulnerable, the command will be executed in the background.")
+        pool = ThreadPoolExecutor(threads)
+        for x in pool.map(worker, URLs_List):
```

Comment:
This should work whether it's a generator (`concurrent.futures.ThreadPoolExecutor`) or not (`multiprocessing.pool.ThreadPool`)